example-app/SantokuApp/src/screens/demo/react-query/useEditTodoDemo.ts

```diff
@@ -0,0 +1,77 @@
+import {useLoadingOverlay} from 'components/overlay';
+import {useCallback, useEffect, useState} from 'react';
+import {useDeleteTodo, useGetTodo, usePutTodo} from 'service/backend';
+
+import {EditTodoDemoScreenProps} from './EditTodoDemoScreen';
+
+export const useEditTodoDemo = ({navigation, route}: EditTodoDemoScreenProps) => {
+  const todoId = route.params.todoId;
+  const [isEdit, setIsEdit] = useState<boolean>(false);
+  const [title, setTitle] = useState<string>();
+  const [description, setDescription] = useState<string>();
+
+  const loadingOverlay = useLoadingOverlay();
+  const todoQuery = useGetTodo(todoId);
+  const todo = todoQuery.data?.data;
+  const putTodo = usePutTodo();
+  const deleteTodo = useDeleteTodo();
+
+  const onEdit = useCallback(() => {
+    setIsEdit(true);
+  }, []);
+
+  const onChangeTitle = useCallback((newTitle: string) => {
+    setTitle(newTitle);
+  }, []);
+
+  const onChangeDescription = useCallback((newDescription: string) => {
+    setDescription(newDescription);
+  }, []);
+
+  const onSave = useCallback(async () => {
+    if (title && description) {
+      try {
+        loadingOverlay.setVisible(true);
+        const data = {title, description};
+        await putTodo.mutateAsync({todoId, data});
+        loadingOverlay.setVisible(false);
+        setIsEdit(false);
+      } catch (e) {
+        loadingOverlay.setVisible(false);
+      }
+    }
+  }, [title, description, loadingOverlay, putTodo, todoId]);
+
+  const onDelete = useCallback(async () => {
+    try {
+      loadingOverlay.setVisible(true);
+      await deleteTodo.mutateAsync({todoId});
+      loadingOverlay.setVisible(false);
+      navigation.goBack();
+    } catch (e) {
+      loadingOverlay.setVisible(false);
+    }
+  }, [deleteTodo, loadingOverlay, navigation, todoId]);
+
+  useEffect(() => {
+    if (!isEdit && todo) {
+      setTitle(todo.title);
+      setDescription(todo.description);
+    }
+  }, [isEdit, todo]);
```

Comment:
ここのイメージは最初に１度だけデータを取得して、以降は使わないという風に感じました。
ここら辺、他でどうやっているかを含めて相談したいです。


```suggestion
  const [initilized, setInitilized] = useState(false);
  useEffect(() => {
    if (!initilized && todo) {
      setTitle(todo.title);
      setDescription(todo.description);
      setInitilized(true);
    }
  }, [initilized, todo]);
```

Reply:
初回だけセットにしてしまうと、編集画面で編集モードにしないままバックグラウンドに移動し、その間に外部でtodoデータが更新された後でアプリがアクティブに戻ってReact Queryによる自動refetchされた際に表示データが更新されなくなりそう。

Reply:
編集モードしかない画面（所謂、編集画面）の場合が一般的に感じたので、デモとしてそちらで用意した方がいいかと感じたのですが、チーム内で話した限りあまり需要もなさそうなので、こちらはクローズします。